### PR TITLE
[spi_device/dv] Update testplan for FW mode

### PR DESCRIPTION
**Pull request by weicaiyang**
Remove mem_ecc and mode as not needed
Update tx/rx_async_fifo_reset

Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

File: hw/ip/spi_device/data/spi_device_testplan.hjson [hunk 1]
```diff
@@ -70,9 +70,34 @@
       tests: ["spi_device_dummy_item_extra_dly"]
     }
     {
-      name: async_fifo_reset
-      desc: '''Reset async fifo when SPI interface is idle
-            TODO: fifo may be fetching data from SRAM? What is the actual usage?'''
+      name: tx_async_fifo_reset
```
Comment (weicaiyang) [inline hw/ip/spi_device/data/spi_device_testplan.hjson:73]: @eunchan @kosta-kojdic please let me know if these 2 entries for async_fifo_reset make sense

File: hw/ip/spi_device/data/spi_device_testplan.hjson [hunk 2]
```diff
@@ -125,21 +150,6 @@
       milestone: V2
       tests: ["spi_device_extreme_fifo_size"]
     }
-    {
-      name: mode
-      desc: '''TODO :only support fw mode now'''
-      milestone: V2
-      tests: []
-    }
-    {
```
Comment (weicaiyang) [inline hw/ip/spi_device/data/spi_device_testplan.hjson:134]: @eunchan I don't think we have ECC in the memory, so I remove this. Please confirm.

Comment (eunchan) [reply hw/ip/spi_device/data/spi_device_testplan.hjson:134]: THanks

File: hw/ip/spi_device/data/spi_device_testplan.hjson (hunk 1 above)
Comment (eunchan) [reply hw/ip/spi_device/data/spi_device_testplan.hjson:73]: Looks good

Comment (kosta-kojdic) [reply hw/ip/spi_device/data/spi_device_testplan.hjson:73]: Looks good, @weicaiyang thank you very much for detailed descriptions of test flows for both tx and rx

**Review by kosta-kojdic — APPROVED** [review]

**Comment by weicaiyang** [comment]
@eunchan can you approve this PR?

**Review by eunchan — APPROVED** [review]

**Pull request merged**